#Travel Insurance

In [25]:
# importing rquired libraries

import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

import plotly.express as px


In [3]:
data = pd.read_csv('travel insurance.csv')
data

Agency    Agency Type Distribution Channel  \
0        CBH  Travel Agency              Offline   
1        CBH  Travel Agency              Offline   
2        CWT  Travel Agency               Online   
3        CWT  Travel Agency               Online   
4        CWT  Travel Agency               Online   
...      ...            ...                  ...   
63321    JZI       Airlines               Online   
63322    JZI       Airlines               Online   
63323    JZI       Airlines               Online   
63324    JZI       Airlines               Online   
63325    JZI       Airlines               Online   

                          Product Name Claim  Duration Destination  Net Sales  \
0                   Comprehensive Plan    No       186    MALAYSIA      -29.0   
1                   Comprehensive Plan    No       186    MALAYSIA      -29.0   
2      Rental Vehicle Excess Insurance    No        65   AUSTRALIA      -49.5   
3      Rental Vehicle Excess Insurance    No        60   AUSTRALIA      -39.6   
4      Rental Vehicle Excess Insurance    No        79       ITALY      -19.8   
...                                ...   ...       ...         ...        ...   
63321                       Basic Plan    No       111       JAPAN       35.0   
63322                       Basic Plan    No        58       CHINA       40.0   
63323                       Basic Plan    No         2    MALAYSIA       18.0   
63324                       Basic Plan    No         3    VIET NAM       18.0   
63325                       Basic Plan    No        22   HONG KONG       26.0   

       Commision (in value) Gender  Age  
0                      9.57      F   81  
1                      9.57      F   71  
2                     29.70    NaN   32  
3                     23.76    NaN   32  
4                     11.88    NaN   41  
...                     ...    ...  ...  
63321                 12.25      M   31  
63322                 14.00      F   40  
63323                  6.30      M   57  
63324                  6.30      M   63  
63325                  9.10      F   35  

[63326 rows x 11 columns]

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63326 entries, 0 to 63325
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Agency                63326 non-null  object 
 1   Agency Type           63326 non-null  object 
 2   Distribution Channel  63326 non-null  object 
 3   Product Name          63326 non-null  object 
 4   Claim                 63326 non-null  object 
 5   Duration              63326 non-null  int64  
 6   Destination           63326 non-null  object 
 7   Net Sales             63326 non-null  float64
 8   Commision (in value)  63326 non-null  float64
 9   Gender                18219 non-null  object 
 10  Age                   63326 non-null  int64  
dtypes: float64(2), int64(2), object(7)
memory usage: 5.3+ MB


In [6]:
X = data.drop(columns= 'Claim')
y = data['Claim']

print(f'shape of X: {X.shape}')
print(f'shape of y: {y.shape}')


shape of X: (63326, 10)
shape of y: (63326,)


## Data Cleaning

In [9]:
X.isnull().mean() * 100

Agency                   0.000000
Agency Type              0.000000
Distribution Channel     0.000000
Product Name             0.000000
Duration                 0.000000
Destination              0.000000
Net Sales                0.000000
Commision (in value)     0.000000
Gender                  71.229827
Age                      0.000000
dtype: float64

Dropping the column `Gender` since 71.2% of the data is missing and on the rest of the columns there is no missing data

In [10]:
X.drop(columns= 'Gender', inplace= True)
print(f'shape of X: {X.shape}')

shape of X: (63326, 9)


## Data Analysis

In [14]:
# unique elements in categorical variables
X.select_dtypes(include= ['object']).nunique()

Agency                   16
Agency Type               2
Distribution Channel      2
Product Name             26
Destination             149
dtype: int64

In [28]:
destination_value_counts = X['Destination'].value_counts()
fig = px.bar(x = destination_value_counts.index, y = destination_value_counts.values)
fig.y
fig.show()